synapse/server_notices/server_notices_manager.py

```diff
@@ -244,7 +244,7 @@ async def _update_notice_user_profile_if_changed(
         assert self.server_notices_mxid is not None
 
         notice_user_data_in_room = await self._message_handler.get_room_data(
-            self.server_notices_mxid,
+            create_requester(self.server_notices_mxid),
```

Comment:
Note that it's the only place outside of tests where I have to create a requester on the fly because I only have a `user_id`